lsst-epo/investigations-client
# [F] Implement initial question/answer functionality

Resolves #41 
Resolves #42 

This PR makes an initial pass at implementing question and answer functionality. Currently it does the following:
- wires up the Questions content block to the API
- connects that content block to the QuestionFactory component, which currently renders SimpleQuestion with either "text" or "select" as the type depending on what's chosen in the CMS. Eventually this will be connected with a wider range of types, once these are added in the backend.
- queries the API when an investigation route is requested and gets any answers that exist for the user.
- creates a React context for tracking the current state of a user's answers. The context's state is initially populated by the query for a user's answers. Once hydrated, the context subscribes to localStorage in the browser for changes.
- wires up individual questions to get their initial value from the context, then updates local storage via the context when they change (or blur for text inputs)
- creates an initial form for saving a user's answers at save points. The form currently sends only the first answer in local storage to Craft as a proof of concept. Once we have support for bulk mutations in the API, we'll send over the whole answer set in one mutation.
- does an initial implementation of token refresh logic for when a user's JWT is expired. This should just run in the background on the server at the necessary time, but we should keep an eye to make sure users aren't getting into weird states.
- updates our GraphQL Codegen config to work with multiple schemas (public, educator, student). I've added some information to the [README](https://github.com/lsst-epo/investigations-client/blob/dj/question-entries/README.md#graphql-codegen--typescript) about our setup generally, as well as how to work with multiple schemas.

## Testing notes

- This PR relies on the `develop` branch in the API repo.
- Make sure to create a unique token for each schema at http://localhost:8080/admin/graphql/tokens, then add the tokens to your `.env.local` file.
- Add some Question entries and a Question block to an Investigation child page in Craft, and you should be all set to test the frontend!
